pkg/test/http_recorder.go

```diff
@@ -239,11 +239,21 @@ func prettifyJSON(s string, mutators ...JSONMutator) string {
 }
 
 func (r *Response) RemoveHeader(key string) {
+	// The http.header `Del` converts the `key` to `CanonicalHeaderKey`, which means
+	// it expects the passed-in parameter `key` to be case insensitive, but `Header` itself should
+	// use canonical keys.
 	r.Header.Del(key)
+	// Delete non canonical header keys like `x-goog-api-client`.
+	delete(r.Header, strings.ToLower(key))
```

Comment:
We _could_ also populate Header with the keys it expects, I.e. use `r.Header.Add` instead of `r.Header[k] = values`

The downside is that our golden tests would then have all the headers canonicalized to lower case, which could be confusing.

Maybe we should stop exporting `Header`, and just have some methods like RemoveHeader, ReplaceHeader etc?  From that perspective, I like what you're doing here!

Reply:
Exporting `header` let us aware of some interesting information, like there's a header about `terraform:kcc-manage` (I plan to change it BTW). 

For mocktesting, +1 that I don't think we really need it.